### PR TITLE
feat: Automated semantically versioned GitHub releases

### PR DESCRIPTION
**Pull request by kenoir**
## What does this change?

This change adds semantic-versioning to releases that will automate publishing releases including binaries as a GitHub release.

## How to test

This has been tested here: https://github.com/kenoir/nest-secrets to avoid polluting this repositories release history during testing.

See https://github.com/kenoir/nest-secrets/releases/tag/v1.2.0 for an example release.

## How can we measure success?

Can these releases be used in another project?

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -21,10 +21,38 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Fetch build
+        uses: actions/download-artifact@v2
```
Comment (nicl) [inline .github/workflows/ci.yaml]: There's a v3 now. Not sure of difference but probably worth bumping.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -21,10 +21,38 @@ jobs:
           go-version: "1.17"
       - name: Run tests
         run: go test .
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
```
Comment (nicl) [inline .github/workflows/ci.yaml]: Worth bumping to v3.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -21,10 +21,38 @@ jobs:
...
+        with:
+          go-version: "1.17"
```
Comment (nicl) [inline .github/workflows/ci.yaml]: Not specific to this PR but worth bumping to 1.19 perhaps?

**Review by nicl — APPROVED** [review]
Some suggestions on bumping versions, but overall looks 👍

File: build.sh [hunk 4]
```diff
@@ -1,5 +1,17 @@
+mkdir -p bin
```
Comment (akash1810) [inline build.sh:5]: Might want a `#!/usr/bin/env bash`, and `set -e` in this file too to ensure the correct interpreter is used and we fail fast.

```suggestion
#!/usr/bin/env bash

set -e

mkdir -p bin
```

**Review by akash1810 — APPROVED** [review]
Added a minor comment, but LGTM 👍🏽 .

**Pull request merged**